liblwgeom/lwgeom_median.c

```diff
@@ -173,11 +198,17 @@ lwmpoint_median(const LWMPOINT* g, double tol, uint32_t max_iter, char fail_if_n
 
 	median = init_guess(points, npoints);
 
+	if (FP_IS_ZERO(median.m))
```

Comment:
Under what condition does our initial guess have a zero m value, and why do we want to return empty in this case?

Reply:
Initial guess can have zero m value if sum of negative values is equal to sum of positive values. 

Let's visualize sum of distances for all cases: 
larger than zero weight, 
`create table median_0_debug as (select ST_SetSRID( ST_MakePoint(x, y), 3857) as geom,  5*sqrt(x^2+y^2) -  sqrt((x-10)^2+y^2)    from generate_series(-100, 100) x, generate_series(-100,100) y);`
![image](https://user-images.githubusercontent.com/810638/34387321-8e70e80a-eb3e-11e7-9c80-75bc420bfb58.png)
 - we have a nice minimum.

Weight of zero,
`create table median_0_debug as (select ST_SetSRID( ST_MakePoint(x, y), 3857) as geom,  sqrt(x^2+y^2) -  sqrt((x-10)^2+y^2)    from generate_series(-100, 100) x, generate_series(-100,100) y);`
![image](https://user-images.githubusercontent.com/810638/34387370-d16b1e14-eb3e-11e7-8b04-1b92d7eeaf2d.png)
- minimum is actually a line in this case. If we add more points, it shows it may be possible to find median:

![image](https://user-images.githubusercontent.com/810638/34387540-d0d699b4-eb3f-11e7-9da5-1a1749270657.png)

Unfortunately, this causes division by zero in centroid code now. A common solution to issues in median now seems to be "fail if asked to fail, empty othewise".

Less than zero:
`create table median_0_debug as (select ST_SetSRID( ST_MakePoint(x, y), 3857) as geom, -1* sqrt(x^2+y^2) -  sqrt((x-10)^2+y^2)    from generate_series(-100, 100) x, generate_series(-100,100) y);`
![image](https://user-images.githubusercontent.com/810638/34387387-eea7b7a8-eb3e-11e7-9f5a-655f7b19e732.png)
 - we have a peak in center, and minimum is in infinity somewhere. It's basically inverse of positive case. We really don't want to iterate to infinity, and unlikely want `POINT(Infinity, Infinity)` in database.